libgalois/include/katana/analytics/ClusteringImplementationBase.h

```diff
@@ -206,35 +214,44 @@ struct ClusteringImplementationBase {
    * Computes the modularity gain of the current cluster assignment
    * without swapping the cluster assignment.
    */
-  uint64_t MaxModularityWithoutSwaps(
-      std::map<uint64_t, uint64_t>& cluster_local_map,
+  void MaxModularityWithoutSwaps(
+      std::map<CommunityIdTy, CommunityIdTy>& cluster_local_map,
       std::vector<EdgeTy>& counter, uint64_t self_loop_wt,
-      CommunityArray& c_info, EdgeTy degree_wt, uint64_t sc, double constant) {
-    uint64_t max_index = sc;  // Assign the intial value as self community
-    double cur_gain = 0;
-    double max_gain = 0;
-    double eix = counter[0] - self_loop_wt;
-    double ax = c_info[sc].degree_wt - degree_wt;
-    double eiy = 0;
-    double ay = 0;
+      CommunityArray& c_info, EdgeTy node_n_degree_wt,
+      GainTy* max_modularity_gain, CommunityIdTy* next_candidate_community,
+      CommunityIdTy curr_comm_id, GainTy constant) {
+    CommunityIdTy max_index = curr_comm_id;
+    GainTy cur_gain = 0;
+    GainTy max_gain = 0;
+
+    //TODO(lhc) edge weight of non-self edges
+    GainTy outgoing_edge_wt = counter[0] - self_loop_wt;
+    GainTy ax = c_info[curr_comm_id].degree_wt - node_n_degree_wt;
+    GainTy neigh_comm_edge_wt = 0;
+    GainTy ay = 0;
 
     auto stored_already = cluster_local_map.begin();
     do {
-      if (sc != stored_already->first) {
+      if (curr_comm_id != stored_already->first) {
         ay = c_info[stored_already->first].degree_wt;  // Degree wt of cluster y
 
-        if (ay < (ax + degree_wt)) {
+        if (ay < (c_info[curr_comm_id].degree_wt)) {
           stored_already++;
           continue;
-        } else if (ay == (ax + degree_wt) && stored_already->first > sc) {
+        } else if (
+            ay == c_info[curr_comm_id].degree_wt &&
+            stored_already->first > curr_comm_id) {
+          // if degree weights are the same, then the big community becomes
```

Comment:
Do you mean the max community id is chosen?

Reply:
yes, that's what he means here

Reply:
Still not clear. "Max" of what? Max ID (because weights are all the same)?